src/game/SpellEffects.cpp

```diff
@@ -5457,33 +5457,21 @@ void Spell::EffectScriptEffect(SpellEffectIndex eff_idx)
                     unitTarget->RemoveAurasAtMechanicImmunity(IMMUNE_TO_ROOT_AND_SNARE_MASK, 30918, true);
                     break;
                 }
-                
-                
-                case 32027:				// Expedition Flare
-			{
-				if (!unitTarget)
-					return;
-
-				uint32 creature_id;
-				uint32 rand = urand(0, 99);
-
-				//summon Expedition Preserver or Expedition Scout 
-				if (rand < 50)
-					creature_id = 18194;
-				else 
-					creature_id = 18126;
-		
-				float angle = unitTarget->GetOrientation();
-			
-				Creature* pCreature = m_caster->SummonCreature(creature_id, 0.0f, 0.0f, 0.0f, angle, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 60000, true);
-
-				if (pCreature->AI())
-					pCreature->AI()->AttackStart(m_caster->getVictim());
-				return;
-
-			}
-                
-        
+                case 32027:                                 // Expedition Flare
+                {
+                    if (!unitTarget)
+                        return;
+
+                    uint32 creature_id = urand(0, 1)
+                                         ? 18194    //summon Expedition Preserver
+                                         : 18126;   //summon Expedition Scout
+
+                    Creature* creature = m_caster->SummonCreature(creature_id, 0.0f, 0.0f, 0.0f, unitTarget->GetOrientation(), TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 60000);
```

Comment:
Have you tried using the dummy effect and look at what m_Targets points to? its dynamic_object_front if i remember correctly, which under some circumstances could give proper location for creature summon, rather than in the caster
if it doesnt work, try to change it so that the creature is summoned next to caster